lib/core/test/unit/Cardano/Wallet/Primitive/ScriptsSpec.hs

```diff
@@ -0,0 +1,383 @@
+    ourSharedKeys === []
+
+prop_scriptUpdatesStateProperly
+    :: AccountXPubWithScripts
+    -> Property
+prop_scriptUpdatesStateProperly (AccountXPubWithScripts accXPub' scripts') = do
+    let (script:_) = scripts'
+    let sciptKeyHashes = retrieveAllVerKeyHashes script
+    let seqState = initializeState accXPub'
+    let (_, seqState') = isShared script seqState
+    let scriptKeyHashesInMap =
+            Set.fromList . map (deriveKeyHash accXPub') <$>
+            Map.lookup (toScriptHash script) (getKnownScripts seqState')
+    scriptKeyHashesInMap === Just (Set.fromList (L.nub sciptKeyHashes))
+
+prop_scriptDiscoveredTwice
+    :: AccountXPubWithScripts
+    -> Property
+prop_scriptDiscoveredTwice (AccountXPubWithScripts accXPub' scripts') = do
+    let (script:_) = scripts'
+    let seqState = initializeState accXPub'
+    let (_, seqState') = isShared script seqState
+    let (_, seqState'') = isShared script seqState'
+    getKnownScripts seqState' === getKnownScripts seqState''
+
+prop_scriptsDiscovered
+    :: AccountXPubWithScripts
+    -> Property
+prop_scriptsDiscovered (AccountXPubWithScripts accXPub' scripts') = do
+    let seqState0 = initializeState accXPub'
+    let seqState = foldr (\script s -> snd $ isShared script s) seqState0 scripts'
+    let scriptHashes = Set.fromList $ Map.keys $ getKnownScripts seqState
+    scriptHashes === Set.fromList (map toScriptHash scripts')
+
+prop_markingDiscoveredVerKeys
+    :: AccountXPubWithScripts
+    -> Property
+prop_markingDiscoveredVerKeys (AccountXPubWithScripts accXPub' scripts') = do
+    let (script:_) = scripts'
+    let sciptKeyHashes = retrieveAllVerKeyHashes script
+    let seqState = initializeState accXPub'
+    let (_, SeqState _ _ _ _ _ verKeyPool) =
+            isShared script seqState
+    let ourKeys = verPoolIndexedKeys verKeyPool
+    let discoveredKeyMap =
+            Map.filterWithKey (\k _ -> k `elem` sciptKeyHashes) ourKeys
+    let addressStatesToCheck =
+            map (\(_, (_, isUsed)) -> isUsed) $ Map.toList discoveredKeyMap
+    L.all (== Used) addressStatesToCheck === True
+
+prop_poolExtension
+    :: AccountXPubWithScriptExtension
+    -> Property
+prop_poolExtension (AccountXPubWithScriptExtension accXPub' scripts') = do
+    scriptHashes === Set.fromList (map toScriptHash scripts')
+  where
+    seqState0 = initializeState accXPub'
+    [script1,script2] = scripts'
+    seqState1 = snd $ isShared script1 seqState0
+    seqState2 = snd $ isShared script2 seqState1
+    scriptHashes = Set.fromList $ Map.keys $ getKnownScripts seqState2
+
+prop_scriptsOutsideNotDiscovered
+    :: AccountXPubWithScriptBeyond
+    -> Property
+prop_scriptsOutsideNotDiscovered (AccountXPubWithScriptBeyond accXPub' script) = do
+    let seqState0 = initializeState accXPub'
+    let seqState = snd $ isShared script seqState0
+    let scriptHashes = Map.keys $ getKnownScripts seqState
+    null scriptHashes === True
```

Comment:
That can be made redundant if adding the check described in my previous comment to the other property.

Reply:
this property is redundant. removed in  https://github.com/input-output-hk/cardano-wallet/pull/2308/commits/ab8567640737c089cb9f15c4be3cfce6dbcadb1a